### ISSUE TITLE
layer_norm_pre_all_gather: kernel assumes zero padding but ttnn.ones fills padding with 1s

### ISSUE THREAD

**Issue by sdjordjevicTT** [tenstorrent/tt-metal#41465]
`ttnn::layer_norm_pre_all_gather` has an implicit contract with its underlying kernel that tile padding is zero. This contract is violated when inputs are created with `ttnn.ones`, causing incorrect statistics.

## Background

The kernel (`layernorm_pre_allgather.cpp`) computes `sum(x)` and `sum(x²)` over the full padded tile width using `PoolType::SUM`. It has no mechanism to distinguish logical width from padded width — it operates on `Wt` tiles unconditionally.

For a logical shape `[1, 1, 37, 72]` (width 72 → padded to 96 = 3 × 32 tiles):

| Input creation | Column 0 (sum(x²)) | Expected |
|---|---|---|
| `ttnn.from_torch(torch.ones(...))` | **72** | 72 ✓ |
| `ttnn.ones(shape=[1,1,37,72])` | **96** | 72 ✗ |

`ttnn.ones` fills the entire padded tile buffer — including the 24 padding elements — with 1s. The kernel then includes those padding elements in the reduction.

## Problem

The TTNN op `layer_norm_pre_all_gather` has an **undocumented contract** that callers must provide zero-padded inputs. This is an implementation detail of the kernel that leaks through the TTNN API layer.

From a compiler perspective, a compiler emitting:
```python
ttnn.layer_norm_pre_all_gather(ttnn.ones(shape=[1,1,37,72], ...))
```
is a perfectly valid op sequence. The compiler operates at the TTNN op level and cannot be expected to reason about kernel-level tile padding assumptions.

## Repro

```python
import ttnn
import torch

device = ttnn.open_mesh_device(
                mesh_shape=ttnn.MeshShape(1,1))

def layer_norm_pre_all_gather(input): 
  var_0 = input[0]
  ttnn_layer_norm_pre_all_gather_0 = ttnn.layer_norm_pre_all_gather(var_0, dtype=ttnn.DataType.BFLOAT16, residual_input_tensor=None, compute_kernel_config=ttnn.WormholeComputeKernelConfig(math_fidelity=ttnn.MathFidelity.HiFi4, fp32_dest_acc_en=True), program_config=None, memory_config=ttnn.MemoryConfig(ttnn.TensorMemoryLayout.INTERLEAVED, ttnn.BufferType.DRAM, None), recip_tensor=None)
  ttnn.deallocate(var_0, False)
  return [ttnn_layer_norm_pre_all_gather_0]

def create_ones_ttnn():
  ttnn_ones_0 = ttnn.ones(shape=ttnn.Shape([1, 1, 37, 72]), dtype=ttnn.DataType.BFLOAT16, layout=ttnn.Layout.TILE, device=device)
  print("TTNN input: ", ttnn_ones_0)
  return [ttnn_ones_0]

def create_ones_torch_to_ttnn():
    torch_ones = torch.ones((1, 1, 37, 72), dtype=torch.bfloat16)
    ttnn_ones_0 = ttnn.from_torch(torch_ones, dtype=ttnn.DataType.BFLOAT16, layout=ttnn.Layout.TILE, device=device)
    print("TTNN input: ", ttnn_ones_0)
    return [ttnn_ones_0]

def main():
  inputs_ones_torch_to_ttnn = create_ones_torch_to_ttnn()
  inputs_ones_ttnn = create_ones_ttnn()
  layer_norm_pre_all_gather_0 = layer_norm_pre_all_gather(inputs_ones_torch_to_ttnn)
  layer_norm_pre_all_gather_1 = layer_norm_pre_all_gather(inputs_ones_ttnn)
  layer_norm_pre_all_gather_0_torch = ttnn.to_torch(layer_norm_pre_all_gather_0[0])
  layer_norm_pre_all_gather_1_torch = ttnn.to_torch(layer_norm_pre_all_gather_1[0])
  print("Output from TTNN with torch input: ", layer_norm_pre_all_gather_0_torch)
  print("Output from TTNN with TTNN input: ", layer_norm_pre_all_gather_1_torch)
  assert torch.allclose(layer_norm_pre_all_gather_0_torch, layer_norm_pre_all_gather_1_torch, rtol=1e-3, atol=1e-3), "Outputs do not match!"
  return 0

if __name__ == '__main__':
  main()
```

## Expected Behavior

One of:
1. `ttnn.ones` (and similar tensor creation ops) should zero-fill tile padding regions, matching the behavior of `ttnn.from_torch`
2. The `layer_norm_pre_all_gather` kernel should accept the logical width and mask padding in the last tile before reduction
3. The TTNN op should validate/document the zero-padding precondition and enforce it

## Impact

Any caller that creates inputs via ops that do not guarantee zero-padded tile regions (e.g. `ttnn.ones`, `ttnn.full`) and feeds them into `layer_norm_pre_all_gather` on shapes that are not multiples of `TILE_WIDTH` (32) will silently get incorrect statistics.
